plugin/src/main/java/com/denizenscript/denizen/events/player/PlayerSpawnChangeScriptEvent.java

```diff
@@ -0,0 +1,84 @@
+package com.denizenscript.denizen.events.player;
+
+import com.denizenscript.denizen.events.BukkitScriptEvent;
+import com.denizenscript.denizen.objects.LocationTag;
+import com.denizenscript.denizen.utilities.implementation.BukkitScriptEntryData;
+import com.denizenscript.denizencore.objects.ObjectTag;
+import com.denizenscript.denizencore.objects.core.ElementTag;
+import com.denizenscript.denizencore.scripts.ScriptEntryData;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import org.bukkit.event.player.PlayerSpawnChangeEvent;
+
+public class PlayerSpawnChangeScriptEvent extends BukkitScriptEvent implements Listener {
+
+    // <--[event]
+    // @Events
+    // player spawn changes
+    //
+    // @Cancellable true
+    //
+    // @Location true
+    //
+    // @Group Player
+    //
+    // @Triggers when a player's spawn point changes.
+    //
+    // @Switch cause:<cause> to only process when the cause for the event matches the input.
+    //
+    // @Context
+    // <context.cause> returns the reason the player's spawn point changed. A list of causes can be found at <@link url https://hub.spigotmc.org/javadocs/spigot/org/bukkit/event/player/PlayerSpawnChangeEvent.Cause.html>.
+    // <context.forced> returns whether this event will persist through source block (bed or respawn anchor) removal.
+    // <context.location> returns a LocationTag of the new respawn location, if any.
+    //
+    // @Determine
+    // "FORCED" to persist even if the bed or respawn anchor that triggered the event is removed.
+    // LocationTag to change the respawn location.
+    //
+    // @Player Always.
+    //
+    // -->
+
+    public PlayerSpawnChangeScriptEvent() {
+        registerCouldMatcher("player spawn changes");
+        registerSwitches("cause");
+        this.<PlayerSpawnChangeScriptEvent>registerTextDetermination("forced", (evt) -> {
+            evt.event.setForced(true);
+        });
```

Comment:
Shouldn't we allow setting this both ways (similarly to `- determine cancelled:false`)?